flaml/ml.py

```diff
@@ -368,6 +368,12 @@ def _eval_estimator(
         pred_start = time.time()
         val_pred_y = get_y_pred(estimator, X_val, eval_metric, obj)
         pred_time = (time.time() - pred_start) / X_val.shape[0]
+
+        if isinstance(estimator, TransformersEstimator):
+            from .nlp.utils import preprocess_labels
+
+            _, y_val = preprocess_labels(X_val, y_val, estimator)

```

Comment:
Should this part be here? When does y_val need preprocessing? Why doesn't TransformerEstimator need preprocessed y_val? Why doesn't _compute_metrics_by_dataset_name need preprocessing?

Reply:
When does y_val need preprocessing? 
>> It needs preprocessing before calling metric_loss_score so could pass the padded labels

Why doesn't TransformerEstimator need preprocessed y_val?
>> TransformersEstimator does preprocess y_val in .fit. Here it calls .predict, which only has access to X_val; so .predict could not preprocess y_val

Why doesn't _compute_metrics_by_dataset_name need preprocessing?
>> _compute_metrics_by_dataset_name takes the argument eval_pred, which contains the preprocessed y_val.

Reply:
Based on what you said, why not preprocess `y_val` before passing to TransformersEstimator in `.fit()`?

Reply:
You mean like in DataTransformer?